src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/ExtractToCodeBehindCodeActionProviderTest.cs

```diff
@@ -0,0 +1,226 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+using Microsoft.AspNetCore.Razor.Language.Extensions;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Xunit;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.CodeActions
+{
+    public class ExtractToCodeBehindCodeActionProviderTest : LanguageServerTestBase
+    {
+        [Fact]
+        public async Task Handle_InvalidFileKind()
+        {
+            // Arrange
+            var documentPath = "c:/Test.razor";
+            var contents = "@page \"/test\"\n@code {}";
```

Comment:
This _may_ cause CI issues. @NTaylorMullen / @ajaybhargavb / @ryanbrandenburg?

```suggestion
            var contents = $"@page \"/test\"{Environment.NewLine}@code {}";
```

Same across the other tests.